src/domain/file/domain.ts

```diff
@@ -37,7 +37,8 @@ const eqFile: Eq<{ id: string }> = pipe(
   eqContramap(it => it.id)
 )
 
-const throwResourceConflictError = (): ResourceConflictError => ResourceConflictError()
+const throwResourceConflictError = (filesIds: FileId[]): ResourceConflictError =>
```

Comment:
The name is misleading because it suggests that the function throws an error. But you don't throw anything here. The function is actually a constructor. So I'd rename it: `createResourceConflictError` or `makeResourceConflictError` or simpler `resourceConflictError`.